app/models/site_status.rb

```diff
@@ -35,6 +39,27 @@ class SiteStatus < ApplicationRecord
     suspended: "S",
   }, _prefix: :status
 
+  aasm column: :status, enum: true do
+    state :new_status, initial: true
+    state :running
+    state :suspended
+    state :discontinued
+
+    after_all_transitions :update_publish_flag
+
+    event :start do
+      transitions from: %i[new_status suspended discontinued], to: :running
+    end
+
+    event :suspend do
+      transitions from: :running, to: :suspended
+    end
+  end
+
+  def update_publish_flag
+    self.publish = (aasm.to_state == :running ? :published : :unpublished)
```

Comment:
Is this change being saved? I don't see where and the only tests I've managed to find (admittedly with only a cursory look) seem to test the `site_status` object without reloading it, which could be prone to giving a false positive.

Reply:
My understanding from [how we're supposed to use the `aasm` library](https://github.com/aasm/aasm/blob/master/README.md#bang-events) is that we don't save explicitly; instead we modify the underlying object and call the `!` method (e.g. `start!`) if we want to auto-save.

I've added `.reload` to the site status specs to ensure that we're not testing stale objects, and the specs are still passing.